src/common/transformations/src/transformations/op_conversions/convert_batch_to_space.cpp

```diff
@@ -4,224 +4,212 @@
-
-        const std::vector<int64_t>& block_values = block_const->cast_vector<int64_t>();
-        const std::vector<int64_t>& crops_end_values = crops_end_const->cast_vector<int64_t>();
-
-        std::vector<int64_t> dispersed_shape(1);
-        dispersed_shape.insert(dispersed_shape.end(), data_shape.begin(), data_shape.end());
-        std::vector<size_t> axes_order(block_values.size() + 1);
-        std::vector<int64_t> squeezed_shape(data_shape.begin(), data_shape.end());
-        if (squeezed_shape.size() > block_values.size()) {
-            return false;
+        const auto data_shape_rank = data.get_partial_shape().rank();
+        if (data_shape_rank.is_dynamic()) {
+            return false;  // beacuse StridedSlice masks are std::vector
         }
 
-        NodeVector new_ops;
-
-        std::shared_ptr<Node> flat_node = data.get_node_shared_ptr();
-        for (size_t block_idx = 1; block_idx < block_values.size(); ++block_idx) {
-            dispersed_shape[0] = block_values[block_idx];
-            dispersed_shape[1] /= block_values[block_idx];
-            const auto out_pattern_1 =
-                opset3::Constant::create(element::i64, Shape{dispersed_shape.size()}, dispersed_shape);
-            const bool special_zero = false;
-            flat_node = std::make_shared<ov::opset3::Reshape>(flat_node, out_pattern_1, special_zero);
-            new_ops.push_back(flat_node);
-
-            size_t val = 1;
-            for (size_t axis_idx = 0; axis_idx <= block_values.size(); ++axis_idx) {
-                if ((block_idx + 1) == axis_idx) {
+        const auto block = batch_to_space->input_value(1);
+        const auto crops_begin = batch_to_space->input_value(2);
+        const auto crops_end = batch_to_space->input_value(3);
+
+        // static data shape rank implies static block shape
+        const auto block_lenght = static_cast<int64_t>(block.get_shape()[0]);
+
+        NodeRegistry rg;
+        const auto zero = rg.make<Constant>(i64, Shape{1}, 0);
+        const auto one = rg.make<Constant>(i64, Shape{1}, 1);
+        const auto two = rg.make<Constant>(i64, Shape{1}, 2);
+        const auto int_max = rg.make<Constant>(i64, Shape{1}, INT_MAX);
+
+        const auto shape_of_data = rg.make<ShapeOf>(data, block.get_element_type());
+        const auto et_zero = rg.make<Constant>(block.get_element_type(), Shape{1}, 0);
+        shared_ptr<Node> dispersed_shape = rg.make<Concat>(OutputVector{et_zero, shape_of_data}, 0);
+        shared_ptr<Node> squeezed_shape = shape_of_data;
+
+        shared_ptr<Node> flat_node = data.get_node_shared_ptr();
+
+        const auto make_concat = [&](OutputVector nodes) {
+            nodes.erase(remove_if(nodes.begin(),
+                                  nodes.end(),
+                                  [](const Output<Node>& n) {
+                                      return n.get_shape()[0] == 0;
```

Comment:
we should check that the shape is static to use get_shape method and that the shape has at least 1 dimension